src/methods/properties.js

```diff
@@ -122,9 +122,9 @@ function InternalUpdatedProperty(realm: Realm, O: ObjectValue, P: PropertyKeyVal
     } else {
       generator.emitPropertyDelete(O, P);
     }
-  } else if (!desc.configurable && desc.enumerable && O === realm.$GlobalObject && desc.value !== undefined) {
+  } else if (flag === "declaration") {
     generator.emitGlobalDeclaration(P, desc.value);
-  } else if (desc.configurable && desc.enumerable && desc.value !== undefined) {
```

Comment:
I think the original code was already slightly broken, in that it didn't detect if the property update also changed any flags. But just checking even fewer flags doesn't make it better...